packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -103,6 +113,20 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkGrammarForUnusedRules(grammar: Grammar, accept: ValidationAcceptor): void {
+        let entryRule = true;
+        for (const rule of grammar.rules) {
+            if (isParserRule(rule) && entryRule) {
+                entryRule = false;
+                continue;
+            }
+            const references = this.references.findReferences(rule);
+            if (references.length === 0) {
+                accept('hint', 'This rule is declared but never referenced', { node: rule, property: 'name', tags: [DiagnosticTag.Unnecessary] });
```

Comment:
In Xtext it's only a hint because we need dummy rules to shape the generated AST metamodel. In the light of #91, we could mark unreachable rules with warnings. Or shall we wait until #91 is actually implemented?

Reply:
I would wait for #91.